gradle-plugin-release-plugin/plugin/src/test/groovy/com/novoda/releaseplugin/internal/ParseChangeLogTaskTest.groovy

```diff
@@ -0,0 +1,54 @@
+package com.novoda.releaseplugin.internal
+
+import com.google.common.io.Resources
+import org.gradle.api.Project
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+
+import static com.google.common.truth.Truth.assertThat
+
+class ParseChangeLogTaskTest {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder()
+
+    private Project project
+
+    @Before
+    void setUp() throws Exception {
+        project = ProjectBuilder
+                .builder()
+                .withProjectDir(temporaryFolder.newFolder())
+                .build()
+    }
+
+    @Test
+    void 'should parse changelog V1'() {
+        def task = project.task('parseReleaseChangeLog', type: ParseChangeLogTask)
+        task.version.set('0.1')
+        def releaseChangelog = temporaryFolder.newFile('v0-1-changelog.md')
```

Comment:
I'm confused with this and it's probably my lack of gradle plugins knowledge but why the output file is prefixed with the version number? Where's that happening in the plugin code?

Reply:
The `ParseChangeLogTaskTest` has the responsibility to extract the changelog for a given release and provide it to anyone interested via it's `OutputFile`.
To do so, it's `OutputFile` is the `releaseChangelog`, the changelog for the given release, e.g. `0.1`. 
In this unit test we specify the `releaseChangelog` as `v0-1-changelog.md`, in production the the `GradleReleasePlugin` configures the `releaseChangelog` with a different name. See `GradleReleasePlugin#configureChangelog`. At this point the plugin also takes the `output` of that task and provides it as input to `printReleaseChangelog`, which then just prints to changelog to the cli.